File: Arrays/Python/Peak_element.py
class Solution:   
    def peakElement(self,arr, n):
        # Code here
        if n==1:
            return 0
        else:
            for i in range(n):
                if i==0 and  arr[i]>arr[i+1]:
                    return i
                elif i==n-1 and arr[i]>arr[i-1]:
                    return i
                elif arr[i]>arr[i-1] and arr[i]>arr[i+1]:
                    return i

File: Arrays/Java/Peak_element.java
class Solution
{
	// Function to find the peak element
	// arr[]: input array
	// n: size of array a[]
	public int peakElement(int[] arr,int n) {
       //add code here.
        int st=0,end=n-1;
        int mid=st+(end-st)/2;
        while(st<end) { 
            if(arr[mid]<arr[mid+1]) {
                st=mid+1;
            }
            else {
                end=mid;
            }
            mid=st+(end-st)/2;
        }
        return st;
    }
}
